[Bug] IntentHandler: Needs a Theme.AppCompat theme

After unsuccessful import

```
ava.lang.RuntimeException: Unable to start activity ComponentInfo{com.ichi2.anki/com.ichi2.anki.IntentHandler}: java.lang.IllegalStateException: You need to use a Theme.AppCompat theme (or descendant) with this activity.
	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:4631)
	at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:4842)
	at android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:113)
	at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:149)
	at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:103)
	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2928)
	at android.os.Handler.dispatchMessage(Handler.java:117)
	at android.os.Looper.loopOnce(Looper.java:205)
	at android.os.Looper.loop(Looper.java:293)
	at android.app.ActivityThread.main(ActivityThread.java:9596)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:586)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1204)
Caused by: java.lang.IllegalStateException: You need to use a Theme.AppCompat theme (or descendant) with this activity.
	at androidx.appcompat.app.AppCompatDelegateImpl.createSubDecor(AppCompatDelegateImpl.java:406)
	at androidx.appcompat.app.AppCompatDelegateImpl.ensureSubDecor(AppCompatDelegateImpl.java:5)
	at androidx.appcompat.app.AppCompatDelegateImpl.setContentView(AppCompatDelegateImpl.java:6)
	at androidx.appcompat.app.AppCompatDialog.setContentView(AppCompatDialog.java:1)
	at androidx.appcompat.app.AlertController.installContent(AlertController.java:7)
	at androidx.appcompat.app.AlertDialog.onCreate(AlertDialog.java:6)
	at android.app.Dialog.dispatchOnCreate(Dialog.java:964)
	at android.app.Dialog.show(Dialog.java:582)
	at androidx.appcompat.app.AlertDialog$Builder.show(AlertDialog.java:5)
	at com.ichi2.utils.ImportUtils$FileImporter.showImportUnsuccessfulDialog(ImportUtils.kt:69)
	at com.ichi2.utils.ImportUtils.showImportUnsuccessfulDialog(ImportUtils.kt:11)
	at com.ichi2.anki.IntentHandler.handleFileImport(IntentHandler.kt:160)
	at com.ichi2.anki.IntentHandler.onCreate$lambda$1(IntentHandler.kt:16)
	at com.ichi2.anki.IntentHandler.b(IntentHandler.kt:1)
	at com.ichi2.anki.p4.run(R8$$SyntheticClass:9)
	at com.ichi2.anki.IntentHandler.performActionIfStorageAccessible(IntentHandler.kt:61)
	at com.ichi2.anki.IntentHandler.onCreate$lambda$0(IntentHandler.kt:16)
	at com.ichi2.anki.IntentHandler.d(IntentHandler.kt:1)
	at com.ichi2.anki.o4.accept(R8$$SyntheticClass:9)
	at com.ichi2.anki.IntentHandler.onCreate(IntentHandler.kt:149)
	at android.app.Activity.performCreate(Activity.java:8488)
	at android.app.Activity.performCreate(Activity.java:8461)
	at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1343)
	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:4602)
	... 12 more
```

<details><summary>Logcat</summary>

```
--------- beginning of main
04-24 10:08:02.955 E/AnkiDroid(30555): CollectionManager/ blocked main thread for 145ms:
04-24 10:08:02.955 E/AnkiDroid(30555): com.ichi2.anki.services.BootService.getColSafe(BootService.kt:7)
04-24 10:10:23.285 E/AnkiDroid(31438): CollectionManager/ blocked main thread for 136ms:
04-24 10:10:23.285 E/AnkiDroid(31438): com.ichi2.anki.services.BootService.getColSafe(BootService.kt:7)
04-24 10:10:27.390 E/AnkiDroid(31438): ImportUtils$FileImporter/ Could not open input stream to intent data
04-24 10:10:27.390 E/AnkiDroid(31438): java.io.FileNotFoundException: open failed: ENOENT (No such file or directory)
04-24 10:10:27.390 E/AnkiDroid(31438): 	at android.database.DatabaseUtils.readExceptionWithFileNotFoundExceptionFromParcel(DatabaseUtils.java:151)
04-24 10:10:27.390 E/AnkiDroid(31438): 	at android.content.ContentProviderProxy.openTypedAssetFile(ContentProviderNative.java:780)
04-24 10:10:27.390 E/AnkiDroid(31438): 	at android.content.ContentResolver.openTypedAssetFileDescriptor(ContentResolver.java:2064)
04-24 10:10:27.390 E/AnkiDroid(31438): 	at android.content.ContentResolver.openAssetFileDescriptor(ContentResolver.java:1879)
04-24 10:10:27.390 E/AnkiDroid(31438): 	at android.content.ContentResolver.openInputStream(ContentResolver.java:1555)
04-24 10:10:27.390 E/AnkiDroid(31438): 	at com.ichi2.utils.ImportUtils$FileImporter.copyFileToCache(ImportUtils.kt:21)
04-24 10:10:27.390 E/AnkiDroid(31438): 	at com.ichi2.utils.ImportUtils$FileImporter.handleContentProviderFile(ImportUtils.kt:234)
04-24 10:10:27.390 E/AnkiDroid(31438): 	at com.ichi2.utils.ImportUtils$FileImporter.handleFileImportInternal(ImportUtils.kt:9)
04-24 10:10:27.390 E/AnkiDroid(31438): 	at com.ichi2.utils.ImportUtils$FileImporter.handleFileImport(ImportUtils.kt:73)
04-24 10:10:27.390 E/AnkiDroid(31438): 	at com.ichi2.utils.ImportUtils.handleFileImport(ImportUtils.kt:16)
04-24 10:10:27.390 E/AnkiDroid(31438): 	at com.ichi2.anki.IntentHandler.handleFileImport(IntentHandler.kt:13)
04-24 10:10:27.390 E/AnkiDroid(31438): 	at com.ichi2.anki.IntentHandler.onCreate$lambda$1(IntentHandler.kt:16)
04-24 10:10:27.390 E/AnkiDroid(31438): 	at com.ichi2.anki.IntentHandler.b(IntentHandler.kt:1)
04-24 10:10:27.390 E/AnkiDroid(31438): 	at com.ichi2.anki.p4.run(R8$$SyntheticClass:9)
04-24 10:10:27.390 E/AnkiDroid(31438): 	at com.ichi2.anki.IntentHandler.performActionIfStorageAccessible(IntentHandler.kt:61)
04-24 10:10:27.390 E/AnkiDroid(31438): 	at com.ichi2.anki.IntentHandler.onCreate$lambda$0(IntentHandler.kt:16)
04-24 10:10:27.390 E/AnkiDroid(31438): 	at com.ichi2.anki.IntentHandler.d(IntentHandler.kt:1)
04-24 10:10:27.390 E/AnkiDroid(31438): 	at com.ichi2.anki.o4.accept(R8$$SyntheticClass:9)
04-24 10:10:27.390 E/AnkiDroid(31438): 	at com.ichi2.anki.IntentHandler.onCreate(IntentHandler.kt:149)
04-24 10:10:27.390 E/AnkiDroid(31438): 	at android.app.Activity.performCreate(Activity.java:8488)
04-24 10:10:27.390 E/AnkiDroid(31438): 	at android.app.Activity.performCreate(Activity.java:8461)
04-24 10:10:27.390 E/AnkiDroid(31438): 	at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1343)
04-24 10:10:27.390 E/AnkiDroid(31438): 	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:4602)
04-24 10:10:27.390 E/AnkiDroid(31438): 	at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:4842)
04-24 10:10:27.390 E/AnkiDroid(31438): 	at android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:113)
04-24 10:10:27.390 E/AnkiDroid(31438): 	at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:149)
04-24 10:10:27.390 E/AnkiDroid(31438): 	at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:103)
04-24 10:10:27.390 E/AnkiDroid(31438): 	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2928)
04-24 10:10:27.390 E/AnkiDroid(31438): 	at android.os.Handler.dispatchMessage(Handler.java:117)
04-24 10:10:27.390 E/AnkiDroid(31438): 	at android.os.Looper.loopOnce(Looper.java:205)
04-24 10:10:27.390 E/AnkiDroid(31438): 	at android.os.Looper.loop(Looper.java:293)
04-24 10:10:27.390 E/AnkiDroid(31438): 	at android.app.ActivityThread.main(ActivityThread.java:9596)
04-24 10:10:27.390 E/AnkiDroid(31438): 	at java.lang.reflect.Method.invoke(Native Method)
04-24 10:10:27.390 E/AnkiDroid(31438): 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:586)
04-24 10:10:27.390 E/AnkiDroid(31438): 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1204)
04-24 10:10:27.390 E/AnkiDroid(31438): 
04-24 10:10:27.390 E/AnkiDroid(31438): java.io.FileNotFoundException: open failed: ENOENT (No such file or directory)
04-24 10:10:27.390 E/AnkiDroid(31438): 	at android.database.DatabaseUtils.readExceptionWithFileNotFoundExceptionFromParcel(DatabaseUtils.java:151)
04-24 10:10:27.390 E/AnkiDroid(31438): 	at android.content.ContentProviderProxy.openTypedAssetFile(ContentProviderNative.java:780)
04-24 10:10:27.390 E/AnkiDroid(31438): 	at android.content.ContentResolver.openTypedAssetFileDescriptor(ContentResolver.java:2064)
04-24 10:10:27.390 E/AnkiDroid(31438): 	at android.content.ContentResolver.openAssetFileDescriptor(ContentResolver.java:1879)
04-24 10:10:27.390 E/AnkiDroid(31438): 	at android.content.ContentResolver.openInputStream(ContentResolver.java:1555)
04-24 10:10:27.390 E/AnkiDroid(31438): 	at com.ichi2.utils.ImportUtils$FileImporter.copyFileToCache(ImportUtils.kt:21)
04-24 10:10:27.390 E/AnkiDroid(31438): 	at com.ichi2.utils.ImportUtils$FileImporter.handleContentProviderFile(ImportUtils.kt:234)
04-24 10:10:27.390 E/AnkiDroid(31438): 	at com.ichi2.utils.ImportUtils$FileImporter.handleFileImportInternal(ImportUtils.kt:9)
04-24 10:10:27.390 E/AnkiDroid(31438): 	at com.ichi2.utils.ImportUtils$FileImporter.handleFileImport(ImportUtils.kt:73)
04-24 10:10:27.390 E/AnkiDroid(31438): 	at com.ichi2.utils.ImportUtils.handleFileImport(ImportUtils.kt:16)
04-24 10:10:27.390 E/AnkiDroid(31438): 	at com.ichi2.anki.IntentHandler.handleFileImport(IntentHandler.kt:13)
04-24 10:10:27.390 E/AnkiDroid(31438): 	at com.ichi2.anki.IntentHandler.onCreate$lambda$1(IntentHandler.kt:16)
04-24 10:10:27.390 E/AnkiDroid(31438): 	at com.ichi2.anki.IntentHandler.b(IntentHandler.kt:1)
04-24 10:10:27.390 E/AnkiDroid(31438): 	at com.ichi2.anki.p4.run(R8$$SyntheticClass:9)
04-24 10:10:27.390 E/AnkiDroid(31438): 	at com.ichi2.anki.IntentHandler.performActionIfStorageAccessible(IntentHandler.kt:61)
04-24 10:10:27.390 E/AnkiDroid(31438): 	at com.ichi2.anki.IntentHandler.onCreate$lambda$0(IntentHandler.kt:16)
04-24 10:10:27.390 E/AnkiDroid(31438): 	at com.ichi2.anki.IntentHandler.d(IntentHandler.kt:1)
04-24 10:10:27.390 E/AnkiDroid(31438): 	at com.ichi2.anki.o4.accept(R8$$SyntheticClass:9)
04-24 10:10:27.390 E/AnkiDroid(31438): 	at com.ichi2.anki.IntentHandler.onCreate(IntentHandler.kt:149)
04-24 10:10:27.390 E/AnkiDroid(31438): 	at android.app.Activity.performCreate(Activity.java:8488)
04-24 10:10:27.390 E/AnkiDroid(31438): 	at android.app.Activity.performCreate(Activity.java:8461)
04-24 10:10:27.390 E/AnkiDroid(31438): 	at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1343)
04-24 10:10:27.390 E/AnkiDroid(31438): 	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:4602)
04-24 10:10:27.390 E/AnkiDroid(31438): 	at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:4842)
04-24 10:10:27.390 E/AnkiDroid(31438): 	at android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:113)
04-24 10:10:27.390 E/AnkiDroid(31438): 	at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:149)
04-24 10:10:27.390 E/AnkiDroid(31438): 	at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:103)
04-24 10:10:27.390 E/AnkiDroid(31438): 	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2928)
04-24 10:10:27.390 E/AnkiDroid(31438): 	at android.os.Handler.dispatchMessage(Handler.java:117)
04-24 10:10:27.390 E/AnkiDroid(31438): 	at android.os.Looper.loopOnce(Looper.java:205)
04-24 10:10:27.390 E/AnkiDroid(31438): 	at android.os.Looper.loop(Looper.java:293)
04-24 10:10:27.390 E/AnkiDroid(31438): 	at android.app.ActivityThread.main(ActivityThread.java:9596)
04-24 10:10:27.390 E/AnkiDroid(31438): 	at java.lang.reflect.Method.invoke(Native Method)
04-24 10:10:27.390 E/AnkiDroid(31438): 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:586)
04-24 10:10:27.390 E/AnkiDroid(31438): 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1204)
04-24 10:10:31.055 E/AnkiDroid(31438): ImportUtils/ showImportUnsuccessfulDialog() message
```
</details>

* https://ankidroid.org/acra/app/1/bug/46237/report/527c4611-d605-48eb-b556-858915720c1b
* HUAWEI MatePad 11 
* 2.16alpha98

--- a/AnkiDroid/src/main/java/com/ichi2/anki/IntentHandler.kt
+++ b/AnkiDroid/src/main/java/com/ichi2/anki/IntentHandler.kt
@@ -65,8 +65,7 @@ class IntentHandler : Activity() {
         // #6157 - We want to block actions that need permissions we don't have, but not the default case
         // as this requires nothing
         val runIfStoragePermissions = Consumer { runnable: Runnable -> performActionIfStorageAccessible(runnable, reloadIntent, action) }
-        val launchType = getLaunchType(intent)
-        when (launchType) {
+        when (getLaunchType(intent)) {
             LaunchType.FILE_IMPORT -> runIfStoragePermissions.accept(Runnable { handleFileImport(intent, reloadIntent, action) })
             LaunchType.SYNC -> runIfStoragePermissions.accept(Runnable { handleSyncIntent(reloadIntent, action) })
             LaunchType.REVIEW -> runIfStoragePermissions.accept(Runnable { handleReviewIntent(intent) })

--- a/AnkiDroid/src/main/java/com/ichi2/anki/IntentHandler.kt
+++ b/AnkiDroid/src/main/java/com/ichi2/anki/IntentHandler.kt
@@ -29,6 +29,7 @@ import com.ichi2.anki.dialogs.DialogHandler.Companion.storeMessage
 import com.ichi2.anki.dialogs.DialogHandlerMessage
 import com.ichi2.anki.servicelayer.ScopedStorageService
 import com.ichi2.anki.services.ReminderService
+import com.ichi2.themes.Themes
 import com.ichi2.themes.Themes.disableXiaomiForceDarkMode
 import com.ichi2.utils.FileUtil
 import com.ichi2.utils.ImportUtils.handleFileImport
@@ -55,6 +56,7 @@ class IntentHandler : Activity() {
         // Note: This is our entry point from the launcher with intent: android.intent.action.MAIN
         Timber.d("onCreate()")
         super.onCreate(savedInstanceState)
+        Themes.setTheme(this)
         disableXiaomiForceDarkMode(this)
         setContentView(R.layout.progress_bar)
         val intent = intent

--- a/AnkiDroid/src/main/java/com/ichi2/utils/ImportUtils.kt
+++ b/AnkiDroid/src/main/java/com/ichi2/utils/ImportUtils.kt
@@ -293,6 +293,7 @@ object ImportUtils {
             AlertDialog.Builder(activity).show {
                 title(text = title)
                 message(text = errorMessage!!)
+                setCancelable(false)
                 positiveButton(R.string.dialog_ok) {
                     if (exitActivity) {
                         AnkiActivity.finishActivityWithFade(activity)

